tools/docker-utils.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const tempy = require('tempy');
+const execa = require('execa');
+const { getUnleashServerVersion } = require('./unleash-server-utils');
+
+const tempDirPath = tempy.directory();
+
+/** @type {string} */
+const dockerfileTemplate = fs.readFileSync(
+  path.resolve(__dirname, '..', 'Dockerfile'),
+  { encoding: 'utf8' },
+);
+
+const fromLineRegExp = /FROM node:.*?\n/g;
+
+/**
+ * Render a Dockerfile with the correct base image.
+ *
+ * @param nodeDockerVersion {string}
+ * @returns {string} the Dockerfile content with the correct base image
+ */
+function renderDockerfile({nodeDockerVersion}) {
+  return dockerfileTemplate.replace(fromLineRegExp, `FROM node:${nodeDockerVersion}`);
+}
+
+const unleashImageName = `unleashorg/unleash-server`;
+
+/**
+ * Renders a Dockerfile to a temporary location on disk
+ * with the specified content.
+ *
+ * @param unleashServerVersion {string} the version of unleash-server that will be included in the image
+ * @param nodeDockerVersion {string} the Node Docker image tag version to base this image on
+ * @returns {Promise<{dockerfilePath: string, unleashDockerTag: string}>}
+ */
+async function createDockerfile({ unleashServerVersion, nodeDockerVersion }) {
+  const unleashDockerTagVersion = `${unleashServerVersion}-node${nodeDockerVersion}`;
+  const unleashDockerTag = `unleashorg/unleash-server:${unleashDockerTagVersion}`;
+
+  const dockerfilePath = path.join(
+    tempDirPath,
+    `Dockerfile.${unleashDockerTagVersion}`,
+  );
+  await fs.promises.writeFile(
+    dockerfilePath,
+    renderDockerfile({ nodeDockerVersion }),
+  );
+
+  return {
+    unleashDockerTag,
+    dockerfilePath,
+  };
+}
+
+/**
+ * Builds an unleash-server Docker image based on a Node docker tag version.
+ *
+ * @param nodeDockerVersion {string} the version of the Node docker image to base this image on
+ * @param unleashServerVersion {string} the version of unleash-server that will be included
+ * @param isDefaultNodeVersion {boolean} whether this version should be used for the tags `latest` and without `-nodeXY-distro`
+ * @returns {Promise<{unleashDockerTags: string[]}>}
+ */
+async function buildDockerImage({
+  nodeDockerVersion,
+  unleashServerVersion,
+  isDefaultNodeVersion,
+}) {
+  console.log(`\nBuilding Docker image based on node:${nodeDockerVersion}`);
+
+  const { dockerfilePath, unleashDockerTag } = await createDockerfile({
+    unleashServerVersion,
+    nodeDockerVersion,
+  });
+
+  const contextDirectory = path.resolve(__dirname, '..');
+
+  await execa(
+    'docker',
+    ['build', '-f', dockerfilePath, '-t', unleashDockerTag, contextDirectory],
+    { stdio: 'inherit' },
+  );
+
+  const additionalUnleashDockerTags = [];
+
+  if (isDefaultNodeVersion) {
+    additionalUnleashDockerTags.push(
+      `${unleashImageName}:${unleashServerVersion}`,
+    );
+    additionalUnleashDockerTags.push(`${unleashImageName}:latest`);
+  }
```

Comment:
here's where the image gets tagged with `latest` and the tag without `-nodeXX-alpine`